### PR TITLE
[wip] - SWP-14: Initial commit for Superdesk Public API Bridge

### PR DESCRIPTION
**Pull request by m038**
- [x] Implement Public api endpoints
- [x] Implement extensible/changeable client
- [x] Restructure classes to not follow Symfony logic
- [x] Implement configuration handling
TODO:
- [x] Write tests
- [x] Re-implement config loading

LATER:
- [ ] Implement asynchronous calls when fetch Package associations
You can find the Superdesk Content API SDK here https://github.com/m038/superdesk-contentapi-sdk-php (will be moved shortly to proper repo).

### CODE REVIEW

File: src/SWP/SuperdeskBridgeBundle/Resources/config/config.yml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+parameters:
```
Comment (ahilles107) [inline src/SWP/SuperdeskBridgeBundle/Resources/config/config.yml]: those are this same as default values in Configuration - so it's not needed. From other side - why we need internall config.yml? We can'd relly on configuration from app? (and provide documnetation for app config.yml) ?

File: src/SWP/SuperdeskBridgeBundle/Resources/config/services.yml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+services:
+
+    swp_sd_bridge.bridge:
+        class: SWP\SuperdeskBridgeBundle\Bridge\Bridge
+        arguments: [@swp_sd_bridge.client, %superdeskbridge%]
+
+    swp_sd_bridge.client:
+        class: SWP\SuperdeskBridgeBundle\Client\Client
+        arguments: [%superdeskbridge%]
```
Comment (ahilles107) [inline src/SWP/SuperdeskBridgeBundle/Resources/config/services.yml]: we should provide special SystemPreferences Service witch will can read from storages (database, custom configuration files) and pass parameters to client from there (with http://symfony.com/doc/current/book/service_container.html#using-the-expression-language) 

example: 

```
arguments:    ["@=service('swp_system_preferences').getParameters('swp_sd_bridge.client')"]

File: src/SWP/SuperdeskBridgeBundle/Tests/Controller/DefaultControllerTest.php [hunk 3]
```diff
@@ -0,0 +1,17 @@
+<?php
```
Comment (ahilles107) [inline src/SWP/SuperdeskBridgeBundle/Tests/Controller/DefaultControllerTest.php]: tests are missing.

**Comment by ahilles107** [comment]
Tests are missing.

File: src/SWP/SuperdeskBridgeBundle/Bridge/Bridge.php [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+namespace SWP\SuperdeskBridgeBundle\Bridge;
+
+use SWP\SuperdeskBridgeBundle\Client\Client;
+use SWP\SuperdeskBridgeBundle\Data\Item;
+use SWP\SuperdeskBridgeBundle\Data\Package;
+
+/**
+ * Superdesk Bridge class.
+ */
+class Bridge
```
Comment (ahilles107) [inline src/SWP/SuperdeskBridgeBundle/Bridge/Bridge.php]: i would move bridge to superdesk/content-api-php-sdk

File: src/SWP/SuperdeskBridgeBundle/Client/Client.php [hunk 5]
```diff
@@ -0,0 +1,144 @@
+<?php
```
Comment (ahilles107) [inline src/SWP/SuperdeskBridgeBundle/Client/Client.php]: i would move client to superdesk/content-api-php-sdk

File: src/SWP/SuperdeskBridgeBundle/Client/ClientInterface.php [hunk 6]
```diff
@@ -0,0 +1,26 @@
+<?php
```
Comment (ahilles107) [inline src/SWP/SuperdeskBridgeBundle/Client/ClientInterface.php]: i would move this to superdesk/content-api-php-sdk

File: src/SWP/SuperdeskBridgeBundle/Data/AbstractDataObject.php [hunk 7]
```diff
@@ -0,0 +1,51 @@
+<?php
```
Comment (ahilles107) [inline src/SWP/SuperdeskBridgeBundle/Data/AbstractDataObject.php]: i would move this to superdesk/content-api-php-sdk

File: src/SWP/SuperdeskBridgeBundle/Data/Item.php [hunk 8]
```diff
@@ -0,0 +1,16 @@
+<?php
```
Comment (ahilles107) [inline src/SWP/SuperdeskBridgeBundle/Data/Item.php]: i would move this to superdesk/content-api-php-sdk

File: src/SWP/SuperdeskBridgeBundle/Data/Package.php [hunk 9]
```diff
@@ -0,0 +1,16 @@
+<?php
```
Comment (ahilles107) [inline src/SWP/SuperdeskBridgeBundle/Data/Package.php]: i would move this to superdesk/content-api-php-sdk

File: src/SWP/SuperdeskBridgeBundle/Exception/BridgeException.php [hunk 10]
```diff
@@ -0,0 +1,18 @@
+<?php
```
Comment (ahilles107) [inline src/SWP/SuperdeskBridgeBundle/Exception/BridgeException.php]: i would move this to superdesk/content-api-php-sdk

**Comment by ahilles107** [comment]
Also check scrutnizer issues - https://scrutinizer-ci.com/g/superdesk/web-renderer/inspections/27f3db6c-bbea-4245-aa9d-91032d28de1f/issues/

File: src/SWP/SuperdeskBridgeBundle/Data/AbstractDataObject.php (hunk 7 above)
Comment (hlmnrmr) [inline src/SWP/SuperdeskBridgeBundle/Data/AbstractDataObject.php]: :+1: a PHP SDK for the Content API, yes!

File: src/SWP/SuperdeskBridgeBundle/Bridge/Bridge.php (hunk 4 above)
Comment (m038) [inline src/SWP/SuperdeskBridgeBundle/Bridge/Bridge.php]: what do you mean by this?

File: src/SWP/SuperdeskBridgeBundle/Resources/config/services.yml (hunk 2 above)
Comment (m038) [inline src/SWP/SuperdeskBridgeBundle/Resources/config/services.yml]: :+1: yeah, that would also remove the need for a config.yml. All required defaults are stored in the class itself anyway.

File: src/SWP/SuperdeskBridgeBundle/Bridge/Bridge.php (hunk 4 above)
Comment (ahilles107) [inline src/SWP/SuperdeskBridgeBundle/Bridge/Bridge.php]: i would create sperate component called superdesk/content-api-php-sdk and keep all mentioned by be in comments classes there and use them in bundle from that component required by composer

Comment (m038) [inline src/SWP/SuperdeskBridgeBundle/Bridge/Bridge.php]: ok, will do that today

File: src/SWP/SuperdeskBridgeBundle/README.md [hunk 11]
```diff
@@ -0,0 +1,14 @@
+Superdesk Content API Bridge bundle
+==========
+
+Configuration options
+
+```yml
+    superdeskbridge:
```
Comment (ahilles107) [inline src/SWP/SuperdeskBridgeBundle/README.md]: it should be swp_superdesk_bridge as is declared here: https://github.com/superdesk/web-renderer/pull/15/files#diff-e368ccdbac45264fc8197fda012fac3fR33

**Comment by ahilles107** [comment]
Please fetch latest master and resolve conflicts + write tests ;)

File: src/SWP/BridgeBundle/Controller/BridgeController.php [hunk 12]
```diff
@@ -0,0 +1,79 @@
...
+        }
+
+        if ($endpointPath === $bridge::SUPERDESK_ENDPOINT_ITEMS) {
+            if (!is_null($objectId)) {
+                $data = $bridge->getItem($objectId);
+            } else {
+                $data = $bridge->getItems($parameters);
+            }
+        } elseif ($endpointPath === $bridge::SUPERDESK_ENDPOINT_PACKAGES) {
+            // TODO: Check if we can do this nicer
```
Comment (ahilles107) [inline src/SWP/BridgeBundle/Controller/BridgeController.php]: You checked that already?

**Comment by m038** [comment]
Closing it, see https://github.com/SuperdeskWebPublisher/SWPBridgeBundle/pull/1

**Comment by ahilles107** [comment]
we could keep here integration code - between bundle loaded as vendor with app.

**Comment by m038** [comment]
What do you mean exactly?

**Comment by ahilles107** [comment]
we will have some config settings for bundle in config.yml - yes? then we will need a PR for this repo ;)

**Pull request closed without merging**